cpp/bench/ann/src/raft/raft_benchmark.cu

```diff
@@ -117,28 +123,25 @@ void parse_search_param(const nlohmann::json& conf,
 }
 #endif
 
-template <typename T, template <typename> class Algo>
-std::unique_ptr<raft::bench::ann::ANN<T>> make_algo(raft::bench::ann::Metric metric,
-                                                    int dim,
-                                                    const nlohmann::json& conf)
+#ifdef RAFT_ANN_BENCH_USE_RAFT_CAGRA
+template <typename T, typename IdxT>
+void parse_build_param(const nlohmann::json& conf,
+                       typename raft::bench::ann::RaftCagra<T, IdxT>::BuildParam& param)
 {
-  typename Algo<T>::BuildParam param;
-  parse_build_param<T>(conf, param);
-  return std::make_unique<Algo<T>>(metric, dim, param);
+  if (conf.contains("index_dim")) {
+    param.graph_degree              = conf.at("index_dim");
+    param.intermediate_graph_degree = param.graph_degree * 2;
+  }
 }
 
-template <typename T, template <typename> class Algo>
-std::unique_ptr<raft::bench::ann::ANN<T>> make_algo(raft::bench::ann::Metric metric,
-                                                    int dim,
-                                                    const nlohmann::json& conf,
-                                                    const std::vector<int>& dev_list)
+template <typename T, typename IdxT>
+void parse_search_param(const nlohmann::json& conf,
+                        typename raft::bench::ann::RaftCagra<T, IdxT>::SearchParam& param)
 {
-  typename Algo<T>::BuildParam param;
-  parse_build_param<T>(conf, param);
-
-  (void)dev_list;
-  return std::make_unique<Algo<T>>(metric, dim, param);
+  param.search_params.max_queries = conf.at("max_batch");
```

Comment:
Just for my understanding: naively I would think that we want to use `max_batch` == `query_set_size` for best perf. In that case setting a lower `max_batch` would only make sense to limit the workspace memory usage. Is this true, or are there any cases where we would want to use a smaller value for `max_batch`?

Reply:
I noticed "search_basic_param.batch_size" is in the conf json files, so I'll remove the "max_batch" setting from CAGRA conf.

> we want to use max_batch == query_set_size for best perf.

Yes, that's true.
However, there are instances where we need to process small batches or even single queries.